test/SILOptimizer/diagnostic_constant_propagation_floats.swift

```diff
@@ -0,0 +1,67 @@
+// RUN: %target-swift-frontend -emit-sil -primary-file %s -o /dev/null -verify
+//
+// These are tests for diagnostics produced by constant propagation pass
+// on floating-point operations
+
+import StdlibUnittest
+
+func testFPToIntConversion() {
+  _blackHole(Int8(-1.28E2))
+  _blackHole(Int8(-1.27E2))
+  _blackHole(Int8(-0))
+  _blackHole(Int8(3.33333))
+  _blackHole(Int8(-2E2)) // expected-error {{floating-point literal '-200' overflows 8-bit signed integer type}}
```

Comment:
These error messages could be a lot better if they printed the actual literal and type that occur in the source. In fact, the user doesn't care that it was a floating-point literal at all, so ideally we should just say something like "This conversion will always fail: `-2E2` overflows `Int8`."

Reply:
@stephentyrone Thanks for the comment.  I have updated the diagnostic messages to display the literal as in the source file, and use the source type, whenever possible. Let me know if the messages look okay now.